Objective:

The following key areas of analysis will be considered:

* Customer Satisfaction Analysis: Analyze the rating data provided by customers to understand the level of satisfaction with different restaurants and cuisines. Identify the factors that contribute to higher customer satisfaction and identify areas for improvement.
    

* Restaurant Demand Analysis: Analyze the order data to identify the popular restaurants and cuisines among customers. Determine the demand patterns based on weekdays and weekends to optimize the availability of delivery personnel and streamline operations.
    

* Food Preparation and Delivery Time Analysis: Analyze the food preparation and delivery time data to identify bottlenecks in the process. Identify restaurants or cuisines that consistently have longer preparation times and work towards reducing them to improve customer experience.
    

* Cost Analysis: Analyze the cost data to identify the average cost of orders and any variations across different restaurants and cuisines. This analysis can help optimize pricing strategies and identify any potential areas for cost reduction.
    

* Customer Retention Analysis: Analyze customer order history to identify patterns of repeat orders and customer loyalty. Identify strategies to improve customer retention and increase customer lifetime value.
    

By focusing on these areas of analysis, the food aggregator company can gain valuable insights to make data-driven decisions and improve various aspects of their business, ultimately leading to enhanced customer satisfaction and sustainable growth.

In [1]:
import pandas as pd
import numpy as np

In [18]:
df = pd.read_csv('food_order.csv')

In [3]:
print (df.head())

   order_id  customer_id            restaurant_name cuisine_type  \
0   1477147       337525                    Hangawi       Korean   
1   1477685       358141  Blue Ribbon Sushi Izakaya     Japanese   
2   1477070        66393                Cafe Habana      Mexican   
3   1477334       106968  Blue Ribbon Fried Chicken     American   
4   1478249        76942           Dirty Bird to Go     American   

   cost_of_the_order day_of_the_week     rating  food_preparation_time  \
0              30.75         Weekend  Not given                     25   
1              12.08         Weekend  Not given                     25   
2              12.23         Weekday          5                     23   
3              29.20         Weekend          3                     25   
4              11.59         Weekday          4                     25   

   delivery_time  
0             20  
1             23  
2             28  
3             15  
4             24  


In [4]:
print (df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1898 entries, 0 to 1897
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   order_id               1898 non-null   int64  
 1   customer_id            1898 non-null   int64  
 2   restaurant_name        1898 non-null   object 
 3   cuisine_type           1898 non-null   object 
 4   cost_of_the_order      1898 non-null   float64
 5   day_of_the_week        1898 non-null   object 
 6   rating                 1898 non-null   object 
 7   food_preparation_time  1898 non-null   int64  
 8   delivery_time          1898 non-null   int64  
dtypes: float64(1), int64(4), object(4)
memory usage: 133.6+ KB
None


In [5]:
print (df.describe())

           order_id    customer_id  cost_of_the_order  food_preparation_time  \
count  1.898000e+03    1898.000000        1898.000000            1898.000000   
mean   1.477496e+06  171168.478398          16.498851              27.371970   
std    5.480497e+02  113698.139743           7.483812               4.632481   
min    1.476547e+06    1311.000000           4.470000              20.000000   
25%    1.477021e+06   77787.750000          12.080000              23.000000   
50%    1.477496e+06  128600.000000          14.140000              27.000000   
75%    1.477970e+06  270525.000000          22.297500              31.000000   
max    1.478444e+06  405334.000000          35.410000              35.000000   

       delivery_time  
count    1898.000000  
mean       24.161749  
std         4.972637  
min        15.000000  
25%        20.000000  
50%        25.000000  
75%        28.000000  
max        33.000000  


In [6]:
# Convert the "Rating" column to numeric
df['rating'] = pd.to_numeric(df['rating'], errors='coerce')

# Verifying the data type of the "Rating" column 
print (df['rating'].dtypes)

float64


In [8]:
# Calculate average rating per restaurant

avg_rating_per_restaurant = df.groupby('restaurant_name')['rating'].mean()

# Identify the top-rated restaurants
top_rated_restaurants = avg_rating_per_restaurant.nlargest(5)

# Print the top-rated restaurants

print("Top-rated restaurants:")
print(top_rated_restaurants)

Top-rated restaurants:
restaurant_name
'wichcraft            5.0
67 Burger             5.0
Anjappar Chettinad    5.0
Asuka Sushi           5.0
Balade                5.0
Name: rating, dtype: float64


In [9]:
# Calculate the number of orders per restaurant
order_count_per_restaurant = df['restaurant_name'].value_counts()

# Identify the popular restaurants
popular_restaurants = order_count_per_restaurant.nlargest(5)

# Print the popular restaurants
print("Popular restaurants:")
print(popular_restaurants)

Popular restaurants:
Shake Shack                  219
The Meatball Shop            132
Blue Ribbon Sushi            119
Blue Ribbon Fried Chicken     96
Parm                          68
Name: restaurant_name, dtype: int64


In [15]:
# Calculate the number of orders per restaurant
order_count_per_restaurant = df['restaurant_name'].value_counts()

# Identify the popular restaurants
popular_restaurants = order_count_per_restaurant.nlargest(5)

# Print the popular restaurants
print("Popular restaurants:")
print(popular_restaurants)

# Calculate the number of orders on weekdays and weekends
order_count_weekday = df[df['day_of_the_week'] == 'Weekday']['restaurant_name'].value_counts()
order_count_weekend = df[df['day_of_the_week'] == 'Weekend']['restaurant_name'].value_counts()

# Identify the restaurants with the highest weekday and weekend demand
weekday_demand_restaurants = order_count_weekday.nlargest(5)
weekend_demand_restaurants = order_count_weekend.nlargest(5)

# Print the restaurants with the highest weekday and weekend demand
if weekday_demand_restaurants.empty:
    print("No restaurants with the highest weekday demand.")
else:
    print("Restaurants with the highest weekday demand:")
    print(weekday_demand_restaurants)

if weekend_demand_restaurants.empty:
    print("No restaurants with the highest weekend demand.")
else:
    print("Restaurants with the highest weekend demand:")
    print(weekend_demand_restaurants)

Popular restaurants:
Shake Shack                  219
The Meatball Shop            132
Blue Ribbon Sushi            119
Blue Ribbon Fried Chicken     96
Parm                          68
Name: restaurant_name, dtype: int64
Restaurants with the highest weekday demand:
Shake Shack                  74
The Meatball Shop            37
Blue Ribbon Sushi            34
Parm                         26
Blue Ribbon Fried Chicken    22
Name: restaurant_name, dtype: int64
Restaurants with the highest weekend demand:
Shake Shack                  145
The Meatball Shop             95
Blue Ribbon Sushi             85
Blue Ribbon Fried Chicken     74
RedFarm Broadway              44
Name: restaurant_name, dtype: int64


In [16]:
# Calculate average food preparation time per restaurant
avg_prep_time_per_restaurant = df.groupby('restaurant_name')['food_preparation_time'].mean()

# Identify the restaurants with the longest average food preparation time
longest_prep_time_restaurants = avg_prep_time_per_restaurant.nlargest(5)

# Print the restaurants with the longest average food preparation time
print("Restaurants with the longest average food preparation time:")
print(longest_prep_time_restaurants)

# Calculate average delivery time per restaurant
avg_delivery_time_per_restaurant = df.groupby('restaurant_name')['delivery_time'].mean()

# Identify the restaurants with the longest average delivery time
longest_delivery_time_restaurants = avg_delivery_time_per_restaurant.nlargest(5)

# Print the restaurants with the longest average delivery time
print("Restaurants with the longest average delivery time:")
print(longest_delivery_time_restaurants)


Restaurants with the longest average food preparation time:
restaurant_name
Cipriani Le Specialita    35.0
Kambi Ramen House         35.0
Klong                     35.0
Sushi Choshi              35.0
Taro Sushi                35.0
Name: food_preparation_time, dtype: float64
Restaurants with the longest average delivery time:
restaurant_name
Sarabeth's West       33.0
Haru Gramercy Park    32.0
Taro Sushi            32.0
Frank Restaurant      31.0
Haandi                30.5
Name: delivery_time, dtype: float64


In [20]:
# Calculate the average cost per restaurant
avg_cost_per_restaurant = df.groupby('restaurant_name')['cost_of_the_order'].mean()

# Identify the restaurants with the highest average cost
highest_average_cost = avg_cost_per_restaurant.nlargest(5)

# Print the restaurants with the highest average cost
print ("Restaurants with the highest average cost")
print (highest_average_cost)

Restaurants with the highest average cost
restaurant_name
Kambi Ramen House      32.930
Emporio                31.430
Bhatti Indian Grill    31.115
Haru Gramercy Park     29.830
Il Bambino             29.250
Name: cost_of_the_order, dtype: float64


In [22]:
# Calculate the average cost on weekdays
avg_cost_weekday = df[df['day_of_the_week'] == 'Weekday']['cost_of_the_order'].mean()

# Calculate the average cost on weekends
avg_cost_weekend = df[df['day_of_the_week'] == 'Weekend']['cost_of_the_order'].mean()

# Print the average cost on weekdays and weekends
print("Average cost on weekdays:", avg_cost_weekday)
print("Average cost on weekends:", avg_cost_weekend)


Average cost on weekdays: 16.311188299817182
Average cost on weekends: 16.574833456698745


In [23]:
# Calculate the number of unique customers
unique_customers = df['customer_id'].nunique()

# Calculate the number of repeat customers
repeat_customers = df['customer_id'].duplicated().sum()

# Calculate the percentage of repeat customers
percentage_repeat_customers = (repeat_customers / unique_customers) * 100

# Print the customer retention analysis results
print("Customer Retention Analysis:")
print("Total Unique Customers:", unique_customers)
print("Number of Repeat Customers:", repeat_customers)
print("Percentage of Repeat Customers:", percentage_repeat_customers)


Customer Retention Analysis:
Total Unique Customers: 1200
Number of Repeat Customers: 698
Percentage of Repeat Customers: 58.166666666666664


* Observations

Customer Satisfaction Analysis:

The top-rated restaurants based on customer ratings are 'wichcraft, 67 Burger, Anjappar Chettinad, Asuka Sushi, and Balade.
Restaurant Demand Analysis:

The popular restaurants based on the order count are Shake Shack, The Meatball Shop, Blue Ribbon Sushi, Blue Ribbon Fried Chicken, and Parm.
Among these popular restaurants, Shake Shack has the highest demand on both weekdays and weekends.
Food Preparation and Delivery Time Analysis:

The restaurants with the longest average food preparation time are Cipriani Le Specialita, Kambi Ramen House, Klong, Sushi Choshi, and Taro Sushi.
The restaurants with the longest average delivery time are Sarabeth's West, Haru Gramercy Park, Taro Sushi, Frank Restaurant, and Haandi.
Cost Analysis:

The restaurants with the highest average cost per order are Kambi Ramen House, Emporio, Bhatti Indian Grill, Haru Gramercy Park, and Il Bambino.
The average cost of orders on weekdays is $16.31, and on weekends, it is $16.57.
Customer Retention Analysis:

Out of the total 1200 unique customers, 698 customers have placed repeat orders, indicating a customer retention rate of 58.17%.


* Recommendations

Customer Satisfaction Analysis:Reach out to the top-rated restaurants, such as 'wichcraft, 67 Burger, Anjappar Chettinad, Asuka Sushi, and Balade, to acknowledge their excellent performance and maintain a strong relationship with them.
Consider featuring these top-rated restaurants prominently in the app to attract more customers and boost overall customer satisfaction.
Analyze customer reviews and feedback for these restaurants to identify specific aspects that contribute to their high ratings. Share these insights with other restaurants to help them improve their offerings and enhance customer satisfaction.

Restaurant Demand Analysis:Work closely with Shake Shack, The Meatball Shop, Blue Ribbon Sushi, Blue Ribbon Fried Chicken, and Parm, as these restaurants have high demand and are popular among customers.
Collaborate with these popular restaurants to ensure efficient order processing, minimize wait times, and optimize the delivery process to enhance customer experience.
Analyze the demand patterns on weekdays and weekends to allocate resources effectively. Consider adjusting staffing levels and delivery personnel availability to meet the fluctuating demand during different days of the week.

Food Preparation and Delivery Time Analysis:Engage with the restaurants that have longer average food preparation and delivery times, such as Cipriani Le Specialita, Kambi Ramen House, Klong, Sushi Choshi, Sarabeth's West, Haru Gramercy Park, Frank Restaurant, and Haandi.
Discuss with these restaurants to identify potential bottlenecks and find ways to improve their processes for faster food preparation and delivery.
Provide training and guidelines to restaurant staff and delivery personnel to optimize efficiency without compromising on food quality or safety.

Cost Analysis:Review the pricing strategy for restaurants with higher average costs per order, including Kambi Ramen House, Emporio, Bhatti Indian Grill, Haru Gramercy Park, and Il Bambino.
Assess the perceived value provided by these restaurants and ensure that the pricing aligns with customer expectations.
Explore opportunities to negotiate better pricing terms with these restaurants to maintain profitability while offering competitive prices to customers.

Customer Retention Analysis:Implement a customer loyalty program or rewards system to incentivize repeat orders and encourage customer retention.
Leverage customer segmentation techniques to identify high-value customers and personalize offers or promotions to enhance their loyalty.
Continuously monitor customer feedback and ratings to identify areas for improvement and address any issues promptly.